### DISCUSSION TITLE
[SUGGESTION] Player view enhancements

### DISCUSSION THREAD

**Discussion by fskaeh** [BLeeEZ/amperfy#320] [Ideas]
Hey, first of all I would like to voice my appreciation for this app. It's not easy to find a nice music client for selfhosted services on iOS!  
To the point: I feel like the fullscreen player view could do with a bit of cleanup in terms of number of icons. Several of those icons would only be occasionally used by most users and could easily stay tucked away in the overflow (3 dots) menu at the bottom of the screen.
- sleep timer button: will only be used once per playing session realistically, since you would set the timer once and then just put your phone down until falling asleep, meaning this button doesn't need to remain onscreen constantly
- playback speed button: unlikely to be used when listening to music, so could be showing only when playing podcasts
- music/podcast switch: ideally this could maybe be replaced by some mechanism that automatically detects the kind of content that is being played and displays the right controls accordingly. No idea how realistic/achievable this is, but even if a manual control is needed, I don't think it's likely that a user would switch back and forth between podcast and music mode often enough to need this button on screen at all times, so it should be hidden in the overflow menu.
- playing queue: a lot of modern music players replace this with a simple scroll down. The playing queue would be below the playback controls, just out of view, and accessible by scrolling. Shuffle/repeat buttons could be placed just above the playing queue to allow users to easily see the playing queue getting updated in response to tapping those buttons.
- airplay button: same idea, you're unlikely to be clicking this repeatedly so it could be tucked away

![IMG_A5DC4E39353B-1](https://github.com/user-attachments/assets/b1a835c1-1fa7-418b-a62d-82680d6afd5b)

- swiping laterally on the album cover to skip to next/previous track: not vital but it is a nice little quality of life feature to have
- background styling: this blur effect looks a little dated. Some apps (like Symfonium among others) tint the background based on some solid or gradient (with a setting) color calculated based on the album cover of the current track (instead of just displayed a heavily blurred version of the cover). It's a very nice effect that makes the app feel polished

**Comment by BLeeEZ** [comment]
Hi, thank you. Some of your proposals sound quite good. I think sleep timer and playback speed can go easily in the "..." menu. There other ones stay there, at least for now. I like the idea of swiping horizontally to trigger play pervious/next.
